File: Problems/206.reverse-linked-list.py
# title: Reverse Linked List
# timestamp: 2024-11-11 22:21:55
# problemUrl: https://leetcode.com/problems/reverse-linked-list/
# memory: 17.4 MB
# runtime: 0 ms

# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head or not head.next:
            return head
        
        prev, curr = None, head
        while curr:
            next = curr.next
            curr.next = prev
            prev = curr
            curr = next
        
        return prev

File: Problems/206.reverse-linked-list.go
// title: Reverse Linked List
// timestamp: 2024-04-18 00:41:35
// problemUrl: https://leetcode.com/problems/reverse-linked-list/
// memory: 2.7 MB
// runtime: 2 ms

/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func reverseList(head *ListNode) *ListNode {
    if head == nil {
        return nil
    }
    var rev *ListNode 
    for head != nil {
        next := head.Next
        head.Next = rev
        rev = head
        head = next
    }
    return rev
}